test/python/quantum_info/states/test_statevector.py

```diff
@@ -1164,6 +1164,29 @@ def test_statevector_draw_latex_regression(self):
         self.assertTrue(latex_string.startswith(" |0\\rangle"))
         self.assertNotIn("|1\\rangle", latex_string)
 
+    def test_statevctor_iter(self):
+        """Test iteration over a state vector"""
+        empty_vector = []
+        dummy_vector = [1, 2, 3]
+        empty_sv = Statevector([])
+        sv = Statevector(dummy_vector)
+
+        # Assert that successive iterations behave as expected, i.e., the
+        # iterator is reset upon each exhaustion of the corresponding
+        # collection of elements.
+        for _ in range(2):
+            self.assertEqual(empty_vector, [x for x in empty_sv])
```

Comment:
Hmm.. the linter seems to [not like](https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=36365&view=logs&j=86d13450-b8b4-5d9a-22f1-9f7e71062a71&t=a7f62dd9-9d78-5db3-3068-aed07887bd76) very much the list comprehension here.
Should this be updated to, e.g., a `for` loop for the sake of the linter? I originally chose a list comprehension, as my feeling is that it captures the use case well.

Reply:
The linter is complaining because the explicit for loop (in this case, inside the comprehension) isn't necessary - we're not doing anything with the values, so we can just write `list(empty_sv)` which is the exact same, but faster to read.

The intent of those rules is usually to try to make simple code look simple, and they're _technically_ a bit faster to execute too (though it shouldn't ever be a bottleneck).

Reply:
Apologies - I should've been clearer in my original comment!

I confirm we're on the same page, in terms of what the reasoning behind that particular linting error is (thank you very much for ensuring it would be conveyed properly). My question was whether you have any opinion on whether that particular linting rule could be considered for removal, as it seems a bit limiting.

I considered alternative framings of the functionality under test, such as `list(iter(sv))` (which I think isn't very readable) and explicitly performing a `for` loop (which I think is a bit verbose).

Based on your response, I understand that the linting rule is here to stay, so I'll update the PR accordingly.